app/src/main/java/org/simple/clinic/teleconsultlog/medicinefrequency/MedicineFrequencyEffectHandler.kt

```diff
@@ -0,0 +1,25 @@
+package org.simple.clinic.teleconsultlog.medicinefrequency
+
+import com.spotify.mobius.rx2.RxMobius
+import com.squareup.inject.assisted.Assisted
+import com.squareup.inject.assisted.AssistedInject
+import io.reactivex.ObservableTransformer
+import org.simple.clinic.util.scheduler.SchedulersProvider
+
+class MedicineFrequencyEffectHandler @AssistedInject constructor(
+    private val schedulersProvider: SchedulersProvider,
+    @Assisted private val uiActions: MedicineFrequencySheetUiActions
+) {
+
+  @AssistedInject.Factory
+  interface Factory {
+    fun create(uiActions: MedicineFrequencySheetUiActions): MedicineFrequencyEffectHandler
+  }
+
+  fun build(): ObservableTransformer<MedicineFrequencyEffect, MedicineFrequencyEvent> {
+    return RxMobius
+        .subtypeEffectHandler<MedicineFrequencyEffect, MedicineFrequencyEvent>()
+        .addConsumer(SetDefaultMedicineFrequency::class.java, { uiActions.setMedicineFrequency(it.medicineFrequency) }, schedulersProvider.ui())
```

Comment:
Oic, I was referring `DrugDuration`, the screen is also setting the drug duration in effect handler. Should that also happen in UiRenderer? I'm trying to figure out the difference between UiActions and UiRendering.

Reply:
Yes, in `DrugDuration` as well it should have been in UiRenderer.

Reply:
I have checked with Vinay and turns out we can use the UiActions for this particular action because prefilling the edit text is a one time action. We are only supposed to use the `ValueChangedCallback` during the Mobius migration. So please ignore this comment.

You can refer to the conversation over [here](https://github.com/simpledotorg/simple-android/pull/1655#discussion_r475603954).